api-examples/rbacv1/main.py

```diff
@@ -0,0 +1,324 @@
+        self.name = name
+        self.email = email
+        self.id_v2 = id_v2
+        self.team_membership = [SLTeamMembership(**t) for t in team_membership]
+
+    def is_member(self, team=""):
+        for tm in self.team_membership:
+            if tm.team_name == tm:
+                return True
+
+
+class SLListUsersResponse:
+    """
+    SLListUsersResponse represents the response for the ListUsers endpoint in ShiftLeft API v4
+    https://docs.shiftleft.io/api/#operation/ListOrgRBACUsers
+    """
+
+    def __init__(self, users=[]):
+        self.users = [SLUser(**u) for u in users]
+
+    def id_for_email(self, user_email=""):
+        for u in self.users:
+            if u.email.lower() == user_email.lower():
+                return u.id_v2
+
+    def user_for_id(self, user_id=""):
+        for u in self.users:
+            if u.id_v2 == user_id:
+                return u
+
+
+class SLTeamInfo:
+    """
+    SLTeamInfo represents the information returned in one item of ListTeams endpoint in ShiftLeft API v4
+    https://docs.shiftleft.io/api/#operation/ListTeams
+    """
+
+    def __init__(self, team_id="", team_name=""):
+        self.team_id = team_id
+        self.team_name = team_name
+
+
+class SLTeams:
+    """
+    SLTeams represents a group of teams, typically of a same organization.
+    """
+
+    def __init__(self, teams=[]):
+        self.teams = [SLTeamInfo(**team) for team in teams]
+
+    def __contains__(self, item):
+        for tm in self.teams:
+            if tm.team_name == item:
+                return True
+
+    def append(self, team):
+        self.teams.append(team)
+
+
+class SLAPIClient:
+    """
+    SLAPIClient handles communications with ShiftLeft API v4 for the purposes of this script.
+    It is very limited and bound to be obsoleted of Schema changes.
+    """
+
+    def __init__(self, access_token="", organization_id=""):
+        self.__access_header = {'Authorization': 'Bearer {}'.format(access_token)}
+        self.__organization_id = organization_id
+
+    def _do_get(self, api_path):
+        u = API_V4_BASE_URL + API_V4_ORG_PATH.format(organization_id=self.__organization_id) + api_path
+        resp = requests.get(u, self.__access_header)
+        handle_status_code(resp)
+        return resp.json().get('response', None)
+
+    def _do_post(self, api_path, payload=None):
+        u = API_V4_BASE_URL + api_path
+        resp = requests.post(u, self.__access_header, payload)
+        handle_status_code(resp)
+        return resp.json().get('response', None)
+
+    def _do_put(self, api_path, payload=None):
+        u = API_V4_BASE_URL + api_path
+        resp = requests.put(u, self.__access_header, payload)
+        handle_status_code(resp)
+        return resp.json().get('response', None)
+
+    def list_users(self):
+        """
+        list_users implements a GET request to https://docs.shiftleft.io/api/#operation/ListOrgRBACUsers
+        :return:
+        """
```

Comment:
instead of the `:return:` you could write this and the following ones like
```suggestion
    def list_users(self) -> SLListUsersResponse:
        """
        list_users implements a GET request to https://docs.shiftleft.io/api/#operation/ListOrgRBACUsers
        """
```

Reply:
Yes, I know, I opted for this without much preference for any of the options and Imho is not worth the change.